gtsfm/utils/geometry_comparisons.py

```diff
@@ -248,3 +248,31 @@ def compute_points_distance_l2(wti1: Optional[Point3], wti2: Optional[Point3]) -
     if wti1 is None or wti2 is None:
         return None
     return np.linalg.norm(wti1 - wti2)
+
+
+def get_points_within_radius_of_cameras(wTi_list: List[Pose3], points_3d: np.ndarray, radius: float = 50) -> Optional[np.ndarray]:
+    """
+    Args:
+        wTi_list: camera poses
+        points_3d: array of shape (N,3) 
+        radius: distance threshold, in meters
+
+    Returns:
+        nearby_points_3d: array of shape (N,3)
+    """
+    if len(wTi_list) == 0 or points_3d.size == 0 or radius < 0:
+        return None
+
+    num_points = points_3d.shape[0]
+    num_poses = len(wTi_list)
+    # each row represents attributes for a single point
+    # each column represents
+    is_nearby_matrix = np.zeros((num_points, num_poses), dtype=bool)
+    for j,wTi in enumerate(wTi_list):
+        is_nearby_matrix[:,j] = np.linalg.norm(points_3d - wTi.translation(), axis=1) < radius
```

Comment:
I'm torn about this. I feel like we should only accept wti_list (translations) since the rotations are unused. But at the same time, you'll need another loop to separate the translations. So, yeah maybe this is better, but just want to point to the alternative.

Reply:
yeah, that's fair. I think a bit simpler to avoid the extra for-loop though